wordpress-mobile/WordPress-Android
# [Lint Warnings] Source Sets: Debug, WordPress and Jetpack - Resolve All Warnings

Parent #18162
Closes #18192

This PR resolves all Lint related warnings for the `Debug`, `WordPress` and `Jetpack` source sets:

-----

Reconfigure Lint (Ignore):

1. [Ignore locale folder lint warnings for jetpack.](https://github.com/wordpress-mobile/WordPress-Android/commit/c272690c89b81589184cf1b56e0bfe5a0d43e90a)

Warnings Resolution List:

1. [Resolve manifest order lint warning for debug.](https://github.com/wordpress-mobile/WordPress-Android/commit/80e895d1a2926b0840c3d2ff0c2a8ca98ae257ce)
2. [Resolve intent filter unique data attributes jetpack warnings.](https://github.com/wordpress-mobile/WordPress-Android/commit/11172969a44dc48f495b97f300d4062520bbf6a2)
3. [Resolve modifier parameter lint warnings for jetpack.](https://github.com/wordpress-mobile/WordPress-Android/commit/bbbddb1b3cad67a2f9c9453b39831075588765c2)

Warnings Suppression List:

1. [Suppress scoped storage lint warning for debug.](https://github.com/wordpress-mobile/WordPress-Android/commit/0ce9dc2a0f86c1ad977c6b0c5948df67237d37be)
2. [Suppress missing application icon lint warning for debug.](https://github.com/wordpress-mobile/WordPress-Android/commit/456eb64bfc7e27d289e919bdda353786397a8991)
3. [Suppress exported content provider lint warning for wordpress.](https://github.com/wordpress-mobile/WordPress-Android/commit/57c07666583162531db6a120f7107dae9c4ce753)
4. [Suppress vector path lint warnings for wordpress/jetpack.](https://github.com/wordpress-mobile/WordPress-Android/commit/1c51abfe6d12f1fbf01fe084b323958643aa41cf)
5. [Suppress missing application icon lint warning for jetpack.](https://github.com/wordpress-mobile/WordPress-Android/commit/015947c175bbd3a75b3bcd202caf828e581fca30)

Documentation List:

1. [Add missing todo for missing firebase instance token refresh.](https://github.com/wordpress-mobile/WordPress-Android/commit/3b659fb01fc2046520436d2543c9779631150672)
2. [Remove unnecessary main section from lint baseline.](https://github.com/wordpress-mobile/WordPress-Android/commit/7d3dda173ff7482512c206ba450a016602c941c3)

-----

PS: @ovitrif I added you as the main reviewer, randomly, since I just wanted someone from the WordPress team to be aware of and sign-off on that change for WPAndroid. I also added the @wordpress-mobile/apps-infrastructure team, but this in done only for monitoring purposes, as such, I am not expecting any active review from that team. Thus, feel free to merge this PR if you deem so.

-----

To test:

- There is nothing much to test here.
- Verifying that all the CI checks are successful should be enough.
- However, if you want to be thorough about reviewing this change, you could quickly smoke test both, the WordPress and Jetpack apps, and see if they are both working as expected.

FYI: As far as I can tell, the only commit that might cause some behavior change might be this 11172969a44dc48f495b97f300d4062520bbf6a2 commit. Thus, you might want to check this one out in a bit more detail/depth. Everything else shouldn't cause any trouble. 

-----

## Regression Notes
1. Potential unintended areas of impact

    - Potential breakage or misbehaviour of the Jetpack related `<intent-filter>` functionality on `DeepLinkingIntentReceiverActivity` and its `android.intent.action.VIEW` action (see 11172969a44dc48f495b97f300d4062520bbf6a2 commit).

6. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To test` section above.

7. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

PR submission checklist:
